### ISSUE TITLE
Font size picker: the Reset button triggers a focus loss

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#57169]
### Description

The font size picker component can optionally render a 'Reset' button.
When clicked, the reset button gets disabled dynamically with the addition of a `disabled` HTML attribute. Current code:

https://github.com/WordPress/gutenberg/blob/34dbbe4226d1d2bcc99c0eb33a48e42a5064edcd/packages/components/src/font-size-picker/index.tsx#L278-L293

As such, when using a keyboard, the button becomes not focusable _while_ it is focused thus triggering a focus loss.

There have been several occurrences so far where contributors used the `disabled` attribute on elements that are supposed to be focused while they get disabled. We need to avoid this pattern at all costs. Cc @ciampo 

Long time ago we established to use another, more accessible, pattern instead:
- Used `aria-disabled` instead of `disabled`.
- Noop the control.

### Step-by-step reproduction instructions

- Go to the storybook at https://wordpress.github.io/gutenberg/?path=/docs/components-fontsizepicker--docs
- Set the `withReset` prop to `true`.
- Click the icon button `Set custom size`.
- Set a custom size.
- Using the keyboard, move focus to the 'Reset' button.
- Activate the button with the Spacebar key.
- Note: the button doesn't seem to work with the Enter key, this needs to be investigated.
- Observe the button gets disabled dynamically.
- Observe focus is lost.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
